zerver/tests/test_send_email.py

```diff
@@ -104,6 +113,64 @@ def test_initialize_connection(self) -> None:
         # 3 more calls to open as we try 3 times before giving up
         self.assertEqual(backend.open.call_count, 6)
 
+    def test_max_connection_lifetime(self) -> None:
+        backend = mock.MagicMock(spec=CustomSMTPEmailBackend)
+        backend.connection = mock.MagicMock(spec=SMTP)
+
+        backend.open.return_value = False
+        backend.connection.noop.return_value = [250]
+
+        # Old connection is open, but we still reopen a new connection because max connection lifetime is 0.
+        with mock.patch.object(backend, "opened_at", timezone_now(), create=True), self.settings(
+            EMAIL_MAX_CONNECTION_LIFETIME_IN_MINUTES=0
+        ):
+            initialize_connection(backend)
```

Comment:
All of these are _probably_ safe -- but we're not using `time_machine` here, so we don't actually know what time it will be when `initialize_connection` checks.  Again, in this case it shouldn't matter what time it is, but it's good practice to _always_ use `time_machine` or similar when we're testing things that are time-dependent.

Reply:
Fixed